adaptive_memory_v4.0.py

```
@@ -0,0 +1,1551 @@
+    # --------------------------------------------------------------------------
+    # Core Pipeline: Inlet (Incoming Message)
+    # --------------------------------------------------------------------------
+    async def inlet(
+        self, body: Dict[str, Any], __event_emitter__=None, __user__=None
+    ) -> Dict[str, Any]:
+        """Process incoming message: Identify user, inject context memories."""
+        if not __user__ or not body.get("messages"):
+            logger.debug("Inlet: No user or messages, skipping.")
+            return body
+
+        logger.debug(f"Inlet called for user {__user__.get('id')}")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)